tests/metrics/map_mem.sh

```diff
@@ -0,0 +1,287 @@
+#!/bin/bash
+
+#  This file is part of cc-oci-runtime.
+#
+#  Copyright (C) 2017 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#  Description of the test:
+#  The objective of this tool is to report the memory map of a process in
+#  a JSON file
+
+declare PROC="$1"
+declare NUM_SAMPLES=10
+declare SAMPLES_FILE="rmap.log"
+declare JSON_file
+declare CC_ID
+
+# Checking env
+function init()
+{
+	req_tools=(
+		pmap
+		awk
+		bc)
+
+	if [ -z "$PROC" ];then
+		echo "error: usage: $0 [process name]"
+		exit 1;
+	fi
+
+	# check available tools
+	echo "check required tools"
+	for cmd in ${req_tools[*]};do
+		command -v "$cmd" &> /dev/null
+		res=$?
+		if [ $res -eq 0 ];then
+			echo "command: $cmd: yes"
+			continue
+		fi
+
+		echo  "error: command $cmd not available"
+	done
+
+	# clean env
+	echo "cleaning env"
+	rm -rf *.log
+	rm -rf *.json
+}
+
+# dump memory avg to JSON file
+function dump_avg_JSON()
+{
+	pss=$1
+	shared=$2
+	private=$3
+	count=$4
+
+	pss_avg=$(echo "scale=2; $pss / $count" | bc)
+	shared_avg=$(echo "scale=2; $shared / $count" | bc -l )
+	private_avg=$(echo "scale=2; $private / $count" | bc -l)
+
+	echo -e "\t\t\"pss\": $pss_avg," >> $JSON_file
+	echo -e "\t\t\"shared\": $shared_avg," >> $JSON_file
+	echo -e "\t\t\"private\": $private_avg" >> $JSON_file
+}
+
+# it creates JSON file
+function create_json()
+{
+	local mapping=""
+	local pss_mem=0
+	local shared_mem=0
+	local private_mem=0
+	local count=0
+
+	# accumulators
+	local shared_sum=0
+	local private_sum=0
+	local pssmem_sum=0
+
+	declare -A memory=(
+		["mapping"]=1
+		["pssmem"]=2
+		["shared"]=3
+		["private"]=4)
+
+	JSON_file="${PROC}.json"
+	output="$(cat "$SAMPLES_FILE" | sort)"
+
+	# start JSON file creation
+	echo "{" > "$JSON_file"
```

Comment:
It would be clearer I think if you collected up all the data you need in variables, then did something like:

```
cat >>"$JSON_file" <<EOT
{
    "foo": "$foo",
    "bar": "$bar"
}
EOT
```